### ISSUE TITLE
load_balancer_profile_outbound_ip_prefix_ids doesn't use a public IP from the prefix range

### ISSUE THREAD

**Issue by abelal83** [Azure/terraform-azurerm-aks#350]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Greenfield/Brownfield provisioning

greenfield

### Terraform Version

1.4.2

### Module Version

6.7.1

### AzureRM Provider Version

3.50.0

### Affected Resource(s)/Data Source(s)

resource "azurerm_kubernetes_cluster" "main"

### Terraform Configuration Files

```hcl

module "aks" {
....
load_balancer_profile_outbound_ip_prefix_ids = ["${module.global.azurerm_subscription.subscription_id}/resourceGroups/pltnetwork-${local.environment_name}-rg/providers/Microsoft.Network/publicIPPrefixes/pltnetwork-${local.environment_name}-001-ippre-ne"]
}
```

### tfvars variables values

```hcl
n/a
```

### Debug Output/Panic Output

```shell
n/a
```

### Expected Behaviour

When public IP for load balancer is created it should use an IP from the public IP prefix resource.

### Actual Behaviour

Deployment creates a public IP with 20.93.105.XXX IP, whereas the prefix IP resource is 20.107.152.XXX/30

**Comment by zioproto** [comment]
Hello @abelal83 
thanks for reporting this issue.

Is possible to provide a minimal configuration to reproduce the issue ?

you provided:
```
module "aks" {
....
load_balancer_profile_outbound_ip_prefix_ids = ["${module.global.azurerm_subscription.subscription_id}/resourceGroups/pltnetwork-${local.environment_name}-rg/providers/Microsoft.Network/publicIPPrefixes/pltnetwork-${local.environment_name}-001-ippre-ne"]
}
```

but this information is not enough for me to reproduce the issue. How does the rest of your config looks like ? I would need at least to know how your `network_profile` block looks like:

https://github.com/Azure/terraform-azurerm-aks/blob/9fbee844c1caa5c7ef5be48750fcdf26053953c5/main.tf#L357-L378

Also, have you checked in the Terraform state that the `load_balancer_profile_outbound_ip_prefix_ids` value is what you expect ?

Thanks

**Comment by abelal83** [comment]
@zioproto thanks for your quick reply.

Our terraform configuration is split across multiple areas, one for networks, another for monitoring etc.. and attempting to add those here would have been quite difficult but appreciate the example I gave doesn't help.

For `load_balancer_profile_outbound_ip_prefix_ids` I pass it the ID for the public IP prefix resource, i.e. `/subscriptions/xxx/resourceGroups/xxxx-dev-rg/providers/Microsoft.Network/publicIPPrefixes/xxxx-dev-001-ippre-ne`. After cluster is created, a public IP appears in the MC-x resource group however, that IP is not from the prefix - it's completely different.

I then tried using `load_balancer_profile_outbound_ip_address_ids` with a public IP resource created from the public IP prefix, it had the same issue, a new public IP was created in MC-X group.

I checked the state file and noticed this value exists

```json
                "load_balancer_profile": [
                  {
                    "effective_outbound_ips": [
                      "/subscriptions/xxxx/resourceGroups/MC_xxx-dev-rg_xxx-dev-001-aks-ne_northeurope/providers/Microsoft.Network/publicIPAddresses/36c84209-6417-4f4d-951d-d58a5c04804d"
                    ],
```
This is unusual in that when I use either `load_balancer_profile_outbound_ip_address_ids` or `load_balancer_profile_outbound_ip_prefix_ids` the value should be one of those.

Hope this helps and many thanks for your kind assistance.
